File: crypto-casino-backend/src/main/java/hu/bme/aut/crypto_casino_backend/dto/stats/QuickStatsResponse.java
package hu.bme.aut.crypto_casino_backend.dto.stats;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuickStatsResponse {

	private BigDecimal vaultBalance;

	private BigDecimal walletBalance;

	private Integer totalGamesPlayed;

	private BigDecimal totalWinnings;

	private Double winRate;

	private BigDecimal biggestWin;

}

File: crypto_casino_android/app/src/main/java/hu/bme/aut/crypto_casino_android/data/model/stats/QuickStatsResponse.kt
package hu.bme.aut.crypto_casino_android.data.model.stats

import java.math.BigDecimal

data class QuickStatsResponse(
    val vaultBalance: BigDecimal,
    val walletBalance: BigDecimal,
    val totalGamesPlayed: Int,
    val totalWinnings: BigDecimal,
    val winRate: Double,
    val biggestWin: BigDecimal
)
